node/custom-db/README.md

```diff
@@ -0,0 +1,231 @@
+
+We are now ready to build the Docker image.
+
+(1) Change into the `docker-only` directory included with this project.
+
+(2) Build the image from the `docker-only` directory.
+![](images/userinput.png)
+>```
+> sudo docker build --pull -t node-img .
+>```
+
+(3) Run the image with Docker.
+![](images/userinput.png)
+>```
+> sudo docker run node-img
+>```
+
+(4) This returns the version of the Oracle DB driver:
+```txt
+3.1.2
+```
+
+That's it. You have a working Docker image.
+
+
+## Convert your Docker image into  an Fn function
+Now let's convert that working Docker image into a function.
+
+**Note:** You can find these files in the `hellodb` directory.
+
+The first step is some changes to the docker file.
+
+```txt
+FROM oraclelinux:7-slim
+
+RUN  yum -y install oracle-release-el7 oracle-nodejs-release-el7 && \
+     yum-config-manager --disable ol7_developer_EPEL && \
+     yum -y install oracle-instantclient19.3-basiclite nodejs && \
+     rm -rf /var/cache/yum
+
+WORKDIR /function
+ADD . /function/
+RUN npm install
+
+CMD exec node func.js
+```
+
+Most of the changes are just cosmetic. The working directory name is "function" rather than "myapp". The launch script is changed from `index.js` to `func.js`  following a normal function template.
+
+Next let's add a `func.yaml` file.
+```js
+schema_version: 20180708
+name: hellodb
+version: 0.0.1
+runtime: docker
+entrypoint: node func.js
+memory: 1024
+```
+
+The function name is `hellodb` so our code should be in a directory with that name. The runtime is set to `docker`. Again notice our node script is `func.js` in this case.
+
+Next, we update the `package.json` file.
+```js
+{
+	"name": "hellodb",
+    "version": "1.0.0",
+	"description": "Node DB Test function",
+	"main": "func.js",
+	"dependencies": {
+		"@fnproject/fdk": ">=0.0.13",
+		"oracledb" : "^3.1"
+	},
+    "author": "Fn Example",
+	"license": "Apache-2.0"
+}
+```
+
+The main change is in dependencies. Here we add the `@fnproject/fdk` dependency to our list. This provides a helper code to run our function on an Fn server. For details on the [Fn Project NPM package click here](https://www.npmjs.com/package/@fnproject/fdk).  Finally, notice we changed the execution script to `func.js`.
+
+For our last preparation step, update the `func.js` file to work with Fn.
+```js
+const fdk=require('@fnproject/fdk');
+const oracledb = require('oracledb');
+
+fdk.handle((input, ctx) => {
+  return {'version': oracledb.versionString};
+})
+```
+A `handle` method is added which is called when a function is invoked. Then, the version is returned as in the previous Docker image. A constant for `fdk` is also added to the script.
+
+With those changes in place perform the following steps to execute the function.
+
+(1) Start the Fn server if it is not already running.
+
+(2) Change into the `hellodb` directory.
+![](images/userinput.png)
+>```
+> cd hellodb
+>```
+
+(3) Next, build your function.
+![](images/userinput.png)
+>```
+> fn -v build
```

Comment:
You could skip the build since deploy will rerun it.

Reply:
Removed.